src/Aspire.Dashboard/DashboardEndpointsBuilder.cs

```diff
@@ -8,12 +8,54 @@
 using Microsoft.AspNetCore.Authentication.OpenIdConnect;
 using Microsoft.AspNetCore.Localization;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.FileProviders;
 using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
 
 namespace Aspire.Dashboard;
 
 public static class DashboardEndpointsBuilder
 {
+    public static void MapDashboardBlazor(this IEndpointRouteBuilder endpoints)
+    {
+        var options = new StaticFileOptions
+        {
+            FileProvider = new ManifestEmbeddedFileProvider(typeof(DashboardEndpointsBuilder).Assembly),
+            OnPrepareResponse = SetCacheHeaders
+        };
+
+        var app = endpoints.CreateApplicationBuilder();
+        app.Use(next => context =>
+        {
+            // Set endpoint to null so the static files middleware will handle the request.
+            context.SetEndpoint(null);
+
+            return next(context);
+        });
```

Comment:
This feel sketchy to me 😄

Reply:
I copied what Blazor itself does: https://github.com/dotnet/aspnetcore/blob/f5853af24e9ee57a5ad26b8609a87c73defc8b33/src/Components/Server/src/Builder/ComponentEndpointRouteBuilderExtensions.cs#L98-L117

Reply:
Is there a reason to not drop the file in the wwwroot folder instead under a different path and render the script tag conditionally based on the runtime you are running on?

It would save you embedding the file and setting up a separate endpoint to serve it.

Reply:
In hindsight a static file probably would have been easier. I was expecting to only have this for .NET 10 and to use the same path as the old embedded file, but I ended up having one file at a new path used all the time.